src/components/Camera/hooks/usePrepareStateForObsEdit.js

```diff
@@ -0,0 +1,109 @@
+  // we want it accessible in the camera's folder, as if the user has taken those photos
+  // via their own camera app).
+  const savePhotosToCameraGallery = useCallback( async uris => {
+    if ( permissionGranted !== "granted" ) { return; }
+    const savedUris = await Promise.all( uris.map( async uri => {
+      // Find original camera URI of each scaled-down photo
+      const cameraUri = originalCameraUrisMap[uri];
+
+      if ( !cameraUri ) {
+        console.error( `Couldn't find original camera URI for: ${uri}` );
+      }
+      logger.info( "savePhotosToCameraGallery, saving cameraUri: ", cameraUri );
+      return CameraRoll.save( cameraUri, { type: "photo", album: "Camera" } );
+    } ) );
+
+    logger.info( "savePhotosToCameraGallery, savedUris: ", savedUris );
+    // Save these camera roll URIs, so later on observation editor can update
+    // the EXIF metadata of these photos, once we retrieve a location.
+    setCameraRollUris( savedUris );
+  }, [originalCameraUrisMap, setCameraRollUris, permissionGranted] );
+
+  const createObsWithCameraPhotos = useCallback( async ( localFilePaths, localTaxon ) => {
+    const newObservation = await Observation.new( );
+    newObservation.observationPhotos = await ObservationPhoto
+      .createObsPhotosWithPosition( localFilePaths, {
+        position: 0,
+        local: true
+      } );
+
+    if ( localTaxon ) {
+      newObservation.taxon = localTaxon;
+      newObservation.owners_identification_from_vision = true;
+    }
+    setObservations( [newObservation] );
+    logger.info(
+      "createObsWithCameraPhotos, calling savePhotosToCameraGallery with paths: ",
+      localFilePaths
+    );
+
+    return savePhotosToCameraGallery( localFilePaths );
+  }, [savePhotosToCameraGallery, setObservations] );
+
+  const createEvidenceForObsEdit = useCallback( async localTaxon => {
+    if ( addEvidence ) {
+      const obsPhotos = await ObservationPhoto
+        .createObsPhotosWithPosition( evidenceToAdd, {
+          position: numOfObsPhotos,
+          local: true
+        } );
+      const updatedCurrentObservation = Observation
+        .appendObsPhotos( obsPhotos, currentObservation );
+      observations[currentObservationIndex] = updatedCurrentObservation;
+      updateObservations( observations );
+      logger.info(
+        "addCameraPhotosToCurrentObservation, calling savePhotosToCameraGallery with paths: ",
+        evidenceToAdd
+      );
+      return savePhotosToCameraGallery( evidenceToAdd );
+    }
+    return createObsWithCameraPhotos( cameraPreviewUris, localTaxon );
+  }, [
+    addEvidence,
+    cameraPreviewUris,
+    createObsWithCameraPhotos,
+    currentObservation,
+    currentObservationIndex,
+    evidenceToAdd,
+    numOfObsPhotos,
+    observations,
+    savePhotosToCameraGallery,
+    updateObservations
+  ] );
+
+  return {
+    createEvidenceForObsEdit: localTaxon => createEvidenceForObsEdit( localTaxon )
```

Comment:
Not sure if you want more feedback here, but FWIW, renaming the file helps, but I think this method is still misleading, since it does a lot more than just creating evidence.